# Week 47 

## Activities

### Activity 3.21

Check your calculation of the determinant of the matrix

$$M= \begin{bmatrix} -1 & 2 & 1 \\ 0 & 2 & 3 \\ 1 & 1 & 4 \end{bmatrix}$$

$$det \space M = -1 \left| \begin{matrix} 2 & 3 \\ 1 & 4 \end{matrix} \right| - 2 \left| \begin{matrix} 0 & 3 \\ 1 & 4 \end{matrix} \right| + 1 \left| \begin{matrix} 0 & 2 \\ 1 & 1 \end{matrix} \right| = -5 + 6 -2 = -1$$

### Activity 6.4

Show that the vectors

$$p=\begin{bmatrix} 1 \\ -2 \end{bmatrix}, \space \space \space q=\begin{bmatrix} -2 \\ 4 \end{bmatrix}$$

$$2p + q = 0$$

### Activity 6.14

This is the method used to find the linear combination given in Example 6.5 for the vectors $v_1$, $v_2$, $v_3$.

$$v_1 = \begin{bmatrix} 1 \\ 2 \\ 3 \end{bmatrix}, \space \space \space v_2 = \begin{bmatrix} 2 \\ 1 \\ 5 \end{bmatrix}, \space \space \space v_3 = \begin{bmatrix} 4 \\ 5 \\ 11 \end{bmatrix}. $$

Find the solution of $a_1v_1+a_2v_2++a_3v_3=0$ to obtain a linear dependence relation.

$$\left[ \begin{array}{ccc|c} 1 & 2 & 4 & 0 \\ 2 & 1 & 5 & 0 \\ 3 & 5 & 11 & 0 \end{array}\right]$$

$$\xrightarrow{R_2-2R_1}R_2$$

$$\left[ \begin{array}{ccc|c} 1 & 2 & 4 & 0 \\ 0 & -3 & -3 & 0 \\ 3 & 5 & 11 & 0 \end{array}\right]$$

$$\xrightarrow{R_3-3R_1}R_3$$

$$\left[ \begin{array}{ccc|c} 1 & 2 & 4 & 0 \\ 0 & -3 & -3 & 0 \\ 0 & -1 & -1 & 0 \end{array}\right]$$

$$\xrightarrow{-1R_3}R_3$$

$$\left[ \begin{array}{ccc|c} 1 & 2 & 4 & 0 \\ 0 & -3 & -3 & 0 \\ 0 & 1 & 1 & 0 \end{array}\right]$$

$$\xrightarrow{-\frac{1}{3}R_2}R_2$$

$$\left[ \begin{array}{ccc|c} 1 & 2 & 4 & 0 \\ 0 & 1 & 1 & 0 \\ 0 & 1 & 1 & 0 \end{array}\right]$$

$$\xrightarrow{R_3-1R_2}R_3$$

$$\left[ \begin{array}{ccc|c} 1 & 2 & 4 & 0 \\ 0 & 1 & 1 & 0 \\ 0 & 0 & 0 & 0 \end{array}\right]$$

$$\xrightarrow{R_1-2R_2}R_1$$

$$\left[ \begin{array}{ccc|c} 1 & 0 & 2 & 0 \\ 0 & 1 & 1 & 0 \\ 0 & 0 & 0 & 0 \end{array}\right]$$

$$x= t\begin{bmatrix}-2 \\ -1 \\ 1 \end{bmatrix}, \space \space t \in \mathbb{R}$$

### Activity 6.30

Carry out the calculation of the determinant. Then verify that $7x+y-3z=0$ is the equation of the plane by checking that the vectors $v_1$, $v_2$, $v_3$ each satisfy this equation.

$$A= \begin{bmatrix} 1 & 2 & 4 \\ 2 & 1 & 5 \\ 3 & 5 & 11 \end{bmatrix}$$

$$det \space A = 1 \left| \begin{matrix} 1 & 5 \\ 5 & 11 \end{matrix} \right| - 2 \left| \begin{matrix} 2 & 5 \\ 3 & 11 \end{matrix} \right| + 4 \left| \begin{matrix} 2 & 1 \\ 3 & 5 \end{matrix} \right| = -14 - 14 + 28 = 0 $$

### Activity 6.59

Let $B$ be the matrix
$$B=\begin{bmatrix} 1 & 1 & 2 & 1 \\ 2 & 0 & 1 & 1 \\ 9 & -1 & 3 & 4 \end{bmatrix}$$

Find a basis of the null space of the matrix $B$ and verify that the rank-nullity theorem:

$$\begin{bmatrix} 1 & 1 & 2 & 1 \\ 2 & 0 & 1 & 1 \\ 9 & -1 & 3 & 4 \end{bmatrix}$$

$$\xrightarrow{R_2-2R_1}R_2$$

$$\begin{bmatrix} 1 & 1 & 2 & 1 \\ 0 & -2 & -3 & -2 \\ 9 & -1 & 3 & 4 \end{bmatrix}$$

$$\xrightarrow{R_2-9R_1}R_2$$

$$\begin{bmatrix} 1 & 1 & 2 & 1 \\ 0 & -2 & -3 & -2 \\ 0 & -10 & -15 & -5 \end{bmatrix}$$

$$\xrightarrow{-\frac{1}{2}R_2}R_2$$

$$\begin{bmatrix} 1 & 1 & 2 & 1 \\ 0 & 1 & \frac{3}{2} & \frac{1}{2} \\ 0 & -10 & -15 & -5 \end{bmatrix}$$

$$\xrightarrow{-\frac{1}{2}R_2}R_2$$

$$\begin{bmatrix} 1 & 1 & 2 & 1 \\ 0 & 1 & \frac{3}{2} & \frac{1}{2} \\ 0 & 0 & 0 & 0 \end{bmatrix}$$

$$\xrightarrow{R_1-1R_2}R_1$$

$$\begin{bmatrix} 1 & 0 & \frac{1}{2} & \frac{1}{2} \\ 0 & 1 & \frac{3}{2} & \frac{1}{2} \\ 0 & 0 & 0 & 0 \end{bmatrix}$$

$$x=t_1\begin{bmatrix} -\frac{1}{2} \\ -\frac{1}{2} \\ 1 \\ 0 \end{bmatrix} + t_2 \begin{bmatrix} -\frac{1}{2} \\ -\frac{1}{2} \\ 0 \\ 1 \end{bmatrix}, \space t \in \mathbb{R}$$

$$rank(B)+nullity(B) = 2 + 2 = 4$$

### Activity 7.30

$$B = \Bigg\{ \begin{pmatrix} 1 \\ 2 \\ -1 \end{pmatrix}, \begin{pmatrix} 2 \\ -1 \\ 4 \end{pmatrix}, \begin{pmatrix} 3 \\ 2 \\ 1 \end{pmatrix} \Bigg\}$$

$$P=\begin{bmatrix} 1 & 2 & 3 \\ 2 & -1 & 2 \\ -1 & 4 & 1 \end{bmatrix}$$

What are $B$ the coordinates of the basis bectors

$$v_1 =\begin{bmatrix} 1 \\ 2 \\ -1 \end{bmatrix}, \space \space \space v_2 = \begin{bmatrix} 2 \\ -1 \\ 4 \end{bmatrix}, \space \space \space v_3 = \begin{bmatrix} 3 \\ 2 \\ 1 \end{bmatrix}, \space \space \space$$

In [8]:
import numpy as np
P = np.matrix([[1, 2, 3], [2, -1, 2], [-1, 4, 1]])
P_inv = np.linalg.inv(P)
print(P_inv)

[[-2.25  2.5   1.75]
 [-1.    1.    1.  ]
 [ 1.75 -1.5  -1.25]]


$$P^{-1} =  \begin{bmatrix} \frac{9}{4} & \frac{5}{2} & \frac{7}{4} \\ -1 & 1 & 1 \\ \frac{7}{4} & -\frac{3}{2} & -\frac{5}{4} \end{bmatrix}$$

In [16]:
v1 = np.transpose(np.matrix([[1, 2, -1]]))
v2 = np.transpose(np.matrix([[2, -1, 4]]))
v3 = np.transpose(np.matrix([[3, 2, 1]]))
print(P_inv*v1)
print(P_inv*v2)
print(P_inv*v3)

[[1.]
 [0.]
 [0.]]
[[-8.8817842e-16]
 [ 1.0000000e+00]
 [ 8.8817842e-16]]
[[0.]
 [0.]
 [1.]]


$$[v_1]_B = \begin{bmatrix} \frac{9}{4} & \frac{5}{2} & \frac{7}{4} \\ -1 & 1 & 1 \\ \frac{7}{4} & -\frac{3}{2} & -\frac{5}{4} \end{bmatrix} \begin{bmatrix} 1 \\ 2 \\ -1 \end{bmatrix} = \begin{bmatrix} 1 \\ 0 \\ 0 \end{bmatrix}$$

$$[v_2]_B = \begin{bmatrix} \frac{9}{4} & \frac{5}{2} & \frac{7}{4} \\ -1 & 1 & 1 \\ \frac{7}{4} & -\frac{3}{2} & -\frac{5}{4} \end{bmatrix} \begin{bmatrix} 2 \\ -1 \\ 4 \end{bmatrix} = \begin{bmatrix} 0 \\ 1 \\ 0 \end{bmatrix}$$

$$[v_3]_B = \begin{bmatrix} \frac{9}{4} & \frac{5}{2} & \frac{7}{4} \\ -1 & 1 & 1 \\ \frac{7}{4} & -\frac{3}{2} & -\frac{5}{4} \end{bmatrix} \begin{bmatrix} 3 \\ 2 \\ 1 \end{bmatrix} = \begin{bmatrix} 0 \\ 0 \\ 1 \end{bmatrix}$$

### Activity 7.43 

In [29]:
A = np.matrix([[1, 3], [-1 ,5]])
P = np.matrix([[1, 3], [1, 1]])
P_inv = np.linalg.inv(P)
print(P_inv)

C = P_inv*(A*P)

print(C)

[[-0.5  1.5]
 [ 0.5 -0.5]]
[[4. 0.]
 [0. 2.]]


$$C = \begin{bmatrix} -\frac{1}{2} & \frac{3}{2} \\ \frac{1}{2} & -\frac{1}{2} \end{bmatrix} \begin{bmatrix} 1 & 3 \\ -1 & 5 \end{bmatrix} \begin{bmatrix} 1 & 3 \\ 1 & 1 \end{bmatrix} = \begin{bmatrix} 4 & 0 \\ 0 & 2 \end{bmatrix}$$

### Activity 7.44

Show that $Av_1 = 4v_1$ and $Av_2 = 2v_2$.

In [31]:
std_basis_vector1 = np.transpose(np.matrix([[1, 1]]))
std_basis_vector2 = np.transpose(np.matrix([[3, 1]]))

print(A*std_basis_vector1)
print(A*std_basis_vector2)

[[4]
 [4]]
[[6]
 [2]]


$$Av_1 = \begin{bmatrix} 1 & 3 \\ -1 & 5 \end{bmatrix} \begin{bmatrix} 1 \\ 1 \end{bmatrix} = \begin{bmatrix} 4 \\ 4 \end{bmatrix}$$

$$Av_2 = \begin{bmatrix} 1 & 3 \\ -1 & 5 \end{bmatrix} \begin{bmatrix} 3 \\ 1 \end{bmatrix} = \begin{bmatrix} 6 \\ 2 \end{bmatrix}$$

## Problems 

### Problem 6.1

In [40]:
L1 = np.matrix([[1, 1], [2, 3]])
L2 = np.matrix([[1, 1, 4], [2, 3, 5]])
L3 = np.matrix([[0, 1], [0, 2]])
L4 = np.matrix([[1, 2, 3], [2, 7, 0], [0, 0, 0]])
L5 = np.matrix([[1, 3, 4], [2, 0, 1], [0, -1 , 2]])

print("L1: " + str(np.linalg.det(L1) == 0))
# print("L2: " + str(np.linalg.det(L2) == 0))
print("L3: " + str(np.linalg.det(L3) == 0))
print("L4: " + str(np.linalg.det(L4) == 0))
print("L5: " + str(np.linalg.det(L5) == 0))

L1: False
L3: True
L4: True
L5: False


### Problem 6.7

Find the coordinates of the vector $\begin{bmatrix} 1 \\ 2 \\ 1 \end{bmatrix}$ with respect to each of the following bases for $\mathbb{R}^3$.

$$B_1=\Bigg\{ \begin{pmatrix} 1 \\ 0 \\ 0 \end{pmatrix}, \begin{pmatrix} 0 \\ 1 \\ 0 \end{pmatrix}, \begin{pmatrix} 0 \\ 0 \\ 1 \end{pmatrix} \Bigg\}, \space \space \space B_2=\Bigg\{ \begin{pmatrix} 1 \\ 1 \\ 1 \end{pmatrix}, \begin{pmatrix} 1 \\ -1 \\ 0 \end{pmatrix}, \begin{pmatrix} 2 \\ -3 \\ -3 \end{pmatrix} \Bigg\}$$

$$P_1 = \begin{bmatrix} 1 & 0 & 0 \\ 0 & 1 & 0 \\ 0 & 0 & 1 \end{bmatrix}, \space \space \space P_2 = \begin{bmatrix} 1 & 1 & 2 \\ 1 & -1 & -3 \\ 1 & 0 & -3 \end{bmatrix}$$

In [47]:
P1 = np.matrix([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
P2 = np.matrix([[1, 1, 2], [1, -1, -3], [1, 0, -3]])

P1_inv = np.linalg.inv(P1)
P2_inv = np.linalg.inv(P2)

print(P2_inv)

v = np.transpose(np.matrix([[1, 2, 1]]))

print(P1_inv*v)
print(P2_inv*v)

[[ 0.6  0.6 -0.2]
 [-0.  -1.   1. ]
 [ 0.2  0.2 -0.4]]
[[1.]
 [2.]
 [1.]]
[[ 1.6]
 [-1. ]
 [ 0.2]]


$$[v]_{B_1} = P_{1}^{-1}v$$

$$P_1 = \begin{bmatrix} 1 & 0 & 0 \\ 0 & 1 & 0 \\ 0 & 0 & 1 \end{bmatrix} \begin{bmatrix} 1 \\ 2 \\ 1 \end{bmatrix} = \begin{bmatrix} 1 \\ 2 \\ 1 \end{bmatrix}$$

$$[v]_{B_2} = P_{2}^{-1}v$$

$$\begin{bmatrix} 1 & 1 & 2 \\ 1 & -1 & -3 \\ 1 & 0 & -3 \end{bmatrix} \begin{bmatrix} 1 \\ 2 \\ 1 \end{bmatrix} = \begin{bmatrix} 1.6 \\ -1 \\ 0.2 \end{bmatrix}$$

### Problem 7.5

If any of the linear transformations $T_i$ given below can be defined, write down a matrix $A_{T_i}$ such that $T_i(x) = A_{T_i}x$. Otherwise, state why $T_i$ is not defined.

$T_1$ : $\mathbb{R^3} \rightarrow \mathbb{R^2}$, where the null space of $T_1$ is the $x$ axis and the range of $T_1$ is the line $y = x$.

$T_2$ : $\mathbb{R^2} \rightarrow \mathbb{R^3}$, such that $N(T_2) = {0}$ and

$$R(T_2) = Lin \Bigg\{ \begin{pmatrix} 1 \\ 0 \\ 1 \end{pmatrix}, \begin{pmatrix} 1 \\ 2 \\ 1 \end{pmatrix}, \begin{pmatrix} 2 \\ 1 \\ 2 \end{pmatrix} \Bigg\}.$$

$T_3$ : $\mathbb{R^2} \rightarrow \mathbb{R^3}$, where null space of $T_3$ is the line $y=2x$ and the range of $T_3$ is the line

$$-\frac{x}{4}=\frac{y}{5}=z$$